### ISSUE TITLE
Tensor documentation hints at differences in graph and eager mode without providing details

### ISSUE THREAD

**Issue by lonerzzz** [tensorflow/tensorflow#49493]
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/Tensor

## Description of issue (what needs changing):

Currently, the description of what a tensor is is cutoff: "A tensor is a multidimensional array of elements represented by a"

Additionally, there is a comment about 

> "Note that during eager execution, you may discover your Tensors are actually of type EagerTensor. This is an internal detail, but it does give you access to a useful function, numpy:"

 This implies that numpy is not available in graph mode. What else isn't available in graph mode? The documentation should describe the differences in capabilities between the two modes - probably on its own page if required.

In my case, I found this out when trying to create a custom loss function, only to find that a number of expected capabilities were not available on the tensor object. It took a long time to find this one line description that hints at the differences. Even then, however, I have no idea what I can use in that mode or where to set the mode so that I can access the data of the tensor.

**Comment by rmothukuru** [comment]
@lonerzzz,
The differences might not be specified in the documentation of [tf.tensor](https://www.tensorflow.org/api_docs/python/tf/Tensor).  Please refer the detailed documentation related to [Eager Execution](https://www.tensorflow.org/guide/eager) and the [Graph Execution](https://www.tensorflow.org/guide/intro_to_graphs) to understand the complete functionalities and the differences between them. Hope this helps. Thanks!

**Comment by lonerzzz** [comment]
Thanks for the reference but the Eager Execution page is still vague in too many ways and does not cover tensors in sufficient detail. For example, this sentence "tf.Variable objects store mutable tf.Tensor-like values accessed during training to make automatic differentiation easier." exists without indicating any detail on the implications of this.

However, I can see serious implications. For example, when I attempt to access a number of tensor functions in a custom loss function that would be present on a tensor, none of them are available. Hence, either the tensor documentation or eager execution documentation needs to be expanded to cover this.

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
